table/sst_file_dumper.cc

```diff
@@ -456,10 +457,23 @@ Status SstFileDumper::ReadSequential(bool print_kv, uint64_t read_num,
       read_options_, moptions_.prefix_extractor.get(),
       /*arena=*/nullptr, /*skip_filters=*/false,
       TableReaderCaller::kSSTDumpTool);
+
+  const Comparator* ucmp = internal_comparator_.user_comparator();
+  size_t ts_sz = ucmp->timestamp_size();
+
+  Slice from_slice = from_key;
+  Slice end_slice = to_key;
+  std::string from_key_buf, end_key_buf;
+  auto [from, end] =
+      MaybeAddTimestampsToRange(from_slice.empty() ? nullptr : &from_slice,
+                                end_slice.empty() ? nullptr : &end_slice, ts_sz,
+                                &from_key_buf, &end_key_buf);
+  if (ts_sz > 0 && (has_from || has_to)) {
```

Comment:
You meant to do something here? also, I'm wondering if `has_from` (and `has_to`) can be removed here. It's equivalent to `!from_key.empty()`, right?

Reply:
oops, I forgot to remove this test code. Thanks for catching it.